src/ssa/ssa_pointed_objects.cpp

```diff
@@ -195,6 +195,22 @@ const exprt symbolic_dereference(const exprt &expr, const namespacet &ns)
 
     return std::move(member);
   }
+  else if(expr.id() == ID_index)
```

Comment:
Have you got a regression test for this?

Reply:
Yes, multiple tests in `regression/arrays` require this (`array_in_dynobj`, `unrolled_empty`, `unrolled_init`).